packages/salesforcedx-vscode-core/src/commands/forceSourceToolingDeploy.ts

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  Command,
+  CompositeCliCommandExecution,
+  CompositeCliCommandExecutor,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import {
+  Row,
+  Table
+} from '@salesforce/salesforcedx-utils-vscode/out/src/output';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
+import { ToolingDeploy, ToolingRetrieveResult } from '../deploys';
+import { nls } from '../messages';
+import { OrgAuthInfo, ToolingDeployParser } from '../util';
+import { SfdxCommandletExecutor } from './util';
+
+export class DeployRetrieveExecutor extends SfdxCommandletExecutor<{}> {
+  public build(sourcePath: string): Command {
+    const commandBuilder = new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_source_deploy_text'))
+      .withArg('force:source:deploy')
+      .withLogName('force_source_tooling_deploy')
+      .withFlag('--sourcepath', sourcePath)
+      .withJson();
+    return commandBuilder.build();
+  }
+
+  public async execute(response: ContinueResponse<string>): Promise<void> {
+    const startTime = process.hrtime();
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const executionWrapper = new CompositeCliCommandExecutor(
+      this.build(response.data)
+    ).execute(cancellationToken);
```

Comment:
It looks like the CLI command is still being built and executed on top of running the tooling deploy logic

Reply:
The CLI command is being built still so it gets surfaced in the output, but from what I understood the `execute()` method of the CompositeCliCommandExecutor doesn't actually spin up a shell process and run the command like the usual **CommandExecutors. It seems to act as more of a wrapper